packages/tasit-identity-contract/src/GnosisSafe.test.js

```diff
@@ -0,0 +1,152 @@
+// from TasitSdk. ProviderFactory is used here as an exception
+// as the clearest way to get a provider
+// in this test suite. Eventually, maybe ProviderFactory may move to
+// some shared helper dir.
+import ProviderFactory from "tasit-action/dist/ProviderFactory";
+
+const { GnosisSafe: GNOSIS_SAFE_ADDRESS } = localAddresses;
+const ERC20_ADDRESS = "0x37E1A58dD465D33263D00185D065Ee36DD34CDb4";
+
+const { utils: ethersUtils } = ethers;
+const { bigNumberify } = ethersUtils;
+
+const ZERO = 0;
+const ONE = bigNumberify(`${1e18}`);
+
+// 100 gwei
+const GAS_PRICE = bigNumberify(`${1e11}`);
+
+describe("GnosisSafe", () => {
+  const config = {
+    provider: {
+      network: "other",
+      provider: "jsonrpc",
+      pollingInterval: 50,
+      jsonRpc: {
+        url: "http://localhost",
+        port: 8545,
+      },
+    },
+    events: {
+      timeout: 2000,
+    },
+  };
+  let gnosisSafeContract;
+  let gnosisSafeUtils;
+  let anaWallet;
+  let snapshotId;
+  let provider;
+  let erc20;
+
+  before("", async () => {
+    ConfigLoader.setConfig(config);
+
+    provider = ProviderFactory.getProvider();
+
+    const anaPrivateKey =
+      "0x11d943d7649fbdeb146dc57bd9cfc80b086bfab2330c7b25651dbaf382392f60";
+
+    anaWallet = createFromPrivateKey(anaPrivateKey);
+
+    // Contract deployment setup with ana (accounts[0]) as the only owner
+    // To change that, edit the file "tasit-contract/3rd-parties/gnosis/scripts/2_deploy_contracts.js"
+    gnosisSafeContract = new GnosisSafe(GNOSIS_SAFE_ADDRESS);
+
+    erc20 = new DetailedERC20(ERC20_ADDRESS);
+  });
+
+  beforeEach("", async () => {
+    snapshotId = await createSnapshot(provider);
+
+    const etherBalance = await provider.getBalance(GNOSIS_SAFE_ADDRESS);
+    expect(`${etherBalance}`).to.equal(`${ZERO}`);
+
+    const erc20Balance = await erc20.balanceOf(GNOSIS_SAFE_ADDRESS);
+    expect(`${erc20Balance}`).to.equal(`${ZERO}`);
+
+    //gnosisSafeContract.removeWallet();
+    //erc20.removeWallet();
+  });
+
+  afterEach("", async () => {
+    await revertFromSnapshot(provider, snapshotId);
+  });
+
+  it("wallet owner should deposit and withdraw some ethers", async () => {
+    anaWallet = anaWallet.connect(provider);
+    const tx = await anaWallet.sendTransaction({
+      to: GNOSIS_SAFE_ADDRESS,
+      value: ONE,
+    });
+    await provider.waitForTransaction(tx.hash);
+
+    const balanceAfterDeposit = await provider.getBalance(GNOSIS_SAFE_ADDRESS);
+    expect(`${balanceAfterDeposit}`).to.equal(`${ONE}`);
+
+    const signers = [anaWallet];
+    const toAddress = anaWallet.address;
+    const value = ONE;
+
+    gnosisSafeContract.setWallet(anaWallet);
+    const execTxAction = await gnosisSafeContract.sendEtherTransaction(
+      signers,
+      toAddress,
+      value
+    );
+    await execTxAction.waitForNonceToUpdate();
+
+    const balanceAfterWithdraw = await provider.getBalance(GNOSIS_SAFE_ADDRESS);
+    expect(`${balanceAfterWithdraw}`).to.equal(`${ZERO}`);
```

Comment:
Should we `expect` anything about the gas used along the way?

Reply:
Oh, yes, sure.
How are you think that we should be done?
I can expose the estimation functions and compare with the gas spent by the transaction, make sense?
Note: To do that will be necessary append gas info to listener data. Make sense?

Reply:
Hmm, I think that makes sense.

You can keep it as a future TODO, though.

Hmm, I can't totally picture why it requires appending gas info to listener data.

Would it be possible to have the user use some gas at a current gas price and assert just that they've used some ether? Or is that not in the spirit of a big test suite where the gas price may need to be free?

Reply:
If we went the route I'm suggesting, we might not need to expose estimation functions. Just make sure _some_ amount was spent on gas

Reply:
Oh, ok.
We can compare `estimatedGas` with `spentGas` (spentGas will be calculated using user balance and gasPrice). Sounds good?

Reply:
Yep, that sounds good!

Reply:
Extracted to: https://github.com/tasitlabs/TasitSDK/issues/235

Reply:
Great, thanks. Resolving this.